crates/ntx-builder/src/state/mod.rs

```diff
@@ -146,13 +180,42 @@ impl State {
             }
 
             self.in_progress.insert(candidate);
-            return TransactionCandidate { account: account.latest_account(), notes }.into();
+            return TransactionCandidate {
+                account: account.latest_account(),
+                notes,
+                chain_tip_header: self.chain_tip_header.clone(),
+                chain_mmr: self.chain_mmr.clone(),
+            }
+            .into();
```

Comment:
We end up cloning `chain_mmr` twice for each transaction: once here, and then once returning it from the `NtxDataStore`. Would be great to figure out how we can avoid one of the clones. Let's create an issue for this (this would be a fairly low-priority issue at this point).